### PR TITLE
README: up-to-date info about how to work on All This Stuff

### PR DESCRIPTION
**Pull request by nfagerlund**
I'm pretty sure this is all accurate (it's kind of my life), but:

- Is it arranged usefully? 
- What did I miss?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,227 @@
...
+
+Earlier instructions for working with this repo said to use `git submodule init` or `git submodule update --init` or `make sync` to activate everything. **Don't do that.** Git commands will take forever to run, and if your `$PS1` includes hints about the current directory's Git status, your entire terminal will slow to a **c r a w l.**
+
+Instead, only init the specific submodules you currently need to work with (`git submodule init ext/providers/aws`), and feel free to de-init them when you're done. De-initting is non-destructive as long as you've committed your changes within the submodule (and preferably pushed your branch) -- Git keeps the repository data cached out of the way, so it dosen't even need to clone the entire repo again the next time you init it.
+
+If you previously activated a hundred submodules and regret it, you can run `git submodule deinit --all` or `make deinit` to start fresh.
+
+### ...Unless You Have to Search Every Page
+
+On rare occasions, you just to search every page on the site for something (usually a broken link). If that's the job, then you do need to init everything. Here's how to make that easier:
```
Comment (armchairlinguist) [inline README.md]:
```suggestion
On rare occasions, you may need to search every page on the site for something (usually a broken link). If that's the job, then you do need to init everything. Here's how to make that easier:
```

**Comment by nfagerlund** [comment]
@armchairlinguist An initial clone does NOT init any submodules! (Haha, though as soon as I wrote that, I had to go make abbbbsolutely sure.) We've all had to de-init, but it's only because we all ran `init` without a target at some point. 

!! THAT's what I forgot, mentioning that those subcommands should always have a target.

**Comment by armchairlinguist** [comment]
Perfect!

File: README.md [hunk 2]
```diff
@@ -1,65 +1,227 @@
...
+
+Avoid running `git rm` on a submodule unless you know what you're doing. You usually want `git submodule deinit` instead.
+
+### Don't Keep Every Submodule Active
+
+Earlier instructions for working with this repo said to use `git submodule init` (with no `<PATH>` argument) or `git submodule update --init` or `make sync` to activate everything. **Don't do that.** Git commands will take forever to run, and if your `$PS1` includes hints about the current directory's Git status, your entire terminal will slow to a **c r a w l.**
+
+Instead, only init the specific submodules you currently need to work with (`git submodule init ext/providers/aws`), and feel free to de-init them when you're done. De-initting is non-destructive as long as you've committed your changes within the submodule (and preferably pushed your branch) -- Git keeps the repository data cached out of the way, so it dosen't even need to clone the entire repo again the next time you init it.
+
+If you previously activated a hundred submodules and regret it, you can run `git submodule deinit --all` or `make deinit` to start fresh.
```
Comment (alisdair) [inline README.md]: Oh my goodness this section is _life changing_ 🙌

Comment (nfagerlund) [reply README.md]: I've been meaning to write this stuff down for ages, IDEK why it took me this long, but I'm glad it's out there now!

**Pull request merged**